### PR TITLE
Add optional visibility_s param to nack_message() method to update message ack deadline

### PR DESCRIPTION
**Pull request by marcin-bakowski-dev**
Add optional visibility_s param to nack_message() method to update message ack deadline

### CODE REVIEW

**Review by maroux — COMMENTED** [review]
There's already extend visibility timeout method, why is this needed?

**Comment by marcin-bakowski-dev** [comment]
> There's already extend visibility timeout method, why is this needed?

I didn't think about it before, `extend_visibility_timeout()` should do the job. I'll try this approach.

There is one case where dependent data from other service is not ready, so I wanted to raise an retry error, but wait couple of seconds before retrying.

**Comment by marcin-bakowski-dev** [comment]
Actually I didn't find a way to handle it with `extend_visibility_timeout()` only.

I tried with decorator:
```
def extend_visibility_timeout_on_retry_exception(visibility_timeout_s: int):
    def decorator(fn):
        @wraps(fn)
        def wrapper(*args, **kwargs):
            try:
                return fn(*args, **kwargs)
            except RetryException:
                metadata = kwargs.get("metadata")
                if metadata and isinstance(metadata, Metadata):
                    logger.debug(
                        "Extending message visibility timeout",
                        visibility_timeout_s=visibility_timeout_s,
                        task=fn.__name__,
                        **metadata.as_dict()
                    )
                    metadata.extend_visibility_timeout(visibility_timeout_s)

        return wrapper

    return decorator
```

and task:
```@taskhawk.task
@transaction.atomic
@extend_visibility_timeout_on_retry_exception(30)
@newrelic.agent.background_task()
def task_example(task_id: str, metadata: Metadata) -> None:
    ...
```

Message final visibility timeout value is overridden in `nack_message()` here: https://github.com/cloudchacho/taskhawk-python/blob/master/taskhawk/backends/base.py#L115, https://github.com/cloudchacho/taskhawk-python/blob/master/taskhawk/backends/gcp.py#L234

**Comment by maroux** [comment]
> Message final visibility timeout value is overridden in `nack_message()` here: https://github.com/cloudchacho/taskhawk-python/blob/master/taskhawk/backends/base.py#L115, https://github.com/cloudchacho/taskhawk-python/blob/master/taskhawk/backends/gcp.py#L234

Oh right, the way to nack a message in pub/sub is to ack with 0 deadline. Hmmm..

I suppose this approach seems reasonable then. One thing to consider: maybe adding a customizable delay in [RetryException](https://github.com/cloudchacho/taskhawk-python/blob/1d297a1c8b76b47289372f054cee4a52c463e72b/taskhawk/exceptions.py#L1) would keep the external API cleaner?

File: taskhawk/backends/aws.py [hunk 1]
```diff
@@ -173,9 +173,11 @@ def process_message(self, queue_message) -> None:
     def delete_message(self, queue_message) -> None:
         queue_message.delete()
 
-    def nack_message(self, queue_message) -> None:
+    def nack_message(self, queue_message, visibility_s: int = 0) -> None:
```
Comment (maroux) [inline taskhawk/backends/aws.py:176]: I wouldn't change this method, just change the behavior of `DelayedRetryException` (or `RetryException` with custom timeout) handling just extends visbility timeout and doesn't nack.

**Comment by marcin-bakowski-dev** [comment]
It's already done in this PR: https://github.com/cloudchacho/taskhawk-python/pull/11

**Pull request closed without merging**